### PR TITLE
[CPU] Fixed enforcebf16 condition for transformation pipeline

### PR DESCRIPTION
**Pull request by a-sidorova**
### Details:
 - *Added check for config parameter `inference_precision`. Before there was only `PluginConfigParams::KEY_ENFORCE_BF16`. It's not enough to know that `enforcebf16` will be really enabled. So some transformations which depend on `enforcebf16` flag, can work incorrectly.*

### Tickets:
 - *N/A*

### CODE REVIEW

**Comment by a-sidorova** [comment]
@dmitry-gorokhov take a look please. This is about Snippets MHA tokenization when the execution is in `bf16`

**Comment by dmitry-gorokhov** [comment]
@a-sidorova Is it possible to add some tests? At least snippets based

**Comment by dmitry-gorokhov** [comment]
@EgorDuplensky Could you please take a look as well? As a fast fix looks good to me, but in future we migh need to refactor this. Maybe it would be a better solution to update plugin config with compile_model properties before transformation pipeline. What is your thoughts on that?

**Comment by EgorDuplensky** [comment]
> @EgorDuplensky Could you please take a look as well? As a fast fix looks good to me, but in future we might need to refactor this. Maybe it would be a better solution to update plugin config with compile_model properties before transformation pipeline. What is your thoughts on that?

Yes, the idea was to merge engine and model configs before doing everything else.
Thus, we will avoid all these extra checks and variables web currently have before executing transformation pipeline.

File: src/tests/functional/plugin/shared/src/snippets/mha.cpp [hunk 1]
```diff
@@ -92,16 +115,41 @@ void MHASelect::generate_inputs(const std::vector<ngraph::Shape>& targetInputSta
...
+
+    auto f = ov::test::snippets::MHAWOTransposeFunction(inputDynamicShapes);
+    function = f.getOriginal();
+
+    configuration.insert(additionalConfig.begin(), additionalConfig.end());
+
+    setInferenceType(prc);
+    inType = outType = prc;
+    if (prc == ov::element::bf16)
+        abs_threshold = 0.3;
```
Comment (maxnick) [inline src/tests/functional/plugin/shared/src/snippets/mha.cpp:152]: Please use a relative threshold. The value should be `0.01` and at least no more than `0.05`

Comment (maxnick) [reply src/tests/functional/plugin/shared/src/snippets/mha.cpp:152]: Agreed to adjust the input data generation in a separate PR.

**Review by maxnick — APPROVED** [review]
LGTM

**Pull request merged**